### ISSUE TITLE
Validate libraries.yml

### ISSUE THREAD

**Issue by chovanecadam** [DannyBen/bashly#439]
### Description

Hello, I was creating my own bashly library with the following `libraries.yml` file:

```yaml
all:
  help: Add all library functions
  files:
    - source: src/config.sh
      target: "%{user_lib_dir}/config.%{user_ext}"
    - source: src/ini.sh
      target: "%{user_lib_dir}/ini.%{user_ext}"
    - source: src/msg.sh
      target: "%{user_lib_dir}/msg.%{user_ext}"
    - soruce: src/prompt.sh
      target: "%{user_lib_dir}/prompt.%{user_ext}"
    - source: src/utility.sh
      target: "%{user_lib_dir}/utility.%{user_ext}"
```

The `bashly add --source ./path/to/lib all` call was failing with error message ` Errno::EISDIR - Is a directory @ io_fread - ../bash-lib/`. The issue lies in a typo -- the fourth item is misspelled (`soruce` instead of `source`). The issue lies [here](https://github.com/DannyBen/bashly/blob/7cf3ef08babcb38c5dca860810660b00dbc01a35/lib/bashly/library.rb#L19), where the code assumes that `source` is set. If it isn't, it calls `file_contents` on the `path`, which fails, because it is a directory.

I think it would be great to either validate the yml file before processing it, or make a better error message. Would you be willing to implement this?

**Comment by DannyBen** [comment]
Agree. Fixing.

**Comment by DannyBen** [comment]
Thanks Adam.
libraries.yml is now validated and released in v1.1.1
